client/src/main.rs

```diff
@@ -115,17 +115,10 @@ async fn run(
 	let cfg_clone = cfg.to_owned();
 	spawn_in_span(shutdown.with_cancel(async move {
 		info!("Bootstraping the DHT with bootstrap nodes...");
-		let bs_result = p2p_clone
+		let _ = p2p_clone
 			.bootstrap_on_startup(&cfg_clone.libp2p.bootstraps)
-			.await;
-		match bs_result {
-			Ok(_) => {
-				info!("Bootstrap done.");
-			},
-			Err(e) => {
-				warn!("Bootstrap process: {e:?}.");
-			},
-		}
+			.await
+			.map_err(|e| warn!("Bootstrap unsuccessful: {e:?}."));
```

Comment:
Using `#` in `warn!` will print error cause along with error text:

```rust
	if let Err(error) = p2p_clone.bootstrap_on_startup(&cfg_clone.libp2p.bootstraps).await {
			warn!("Bootstrap unsuccessful: {error:#}");
	}
```

Same applies to the `crawler` and `fat` clients.